common/beerocks/bwl/dwpal/mon_wlan_hal_dwpal.cpp

```diff
@@ -677,94 +677,92 @@ bool mon_wlan_hal_dwpal::update_vap_stats(const std::string &vap_iface_name, SVa
-    // LOG(DEBUG) << "numOfValidArgs[5]= " << numOfValidArgs[5] << " ShortTermRSSIAverage= " << ShortTermRSSIAverage;
-    // LOG(DEBUG) << "numOfValidArgs[6]= " << numOfValidArgs[6] << " SNR= " << SNR;
-    // LOG(DEBUG) << "numOfValidArgs[0]= " << numOfValidArgs[7] << " Active= " << Active;
-    // LOG(DEBUG) << "numOfValidArgs[7]= " << numOfValidArgs[8] << " LastDataDownlinkRate= " << LastDataDownlinkRate;
-    // LOG(DEBUG) << "numOfValidArgs[8]= " << numOfValidArgs[9] << " LastDataUplinkRate= " << LastDataUplinkRate;
-    /* End of TEMP: Traces... */
-
-    for (uint8_t i = 0; i < (sizeof(numOfValidArgs) / sizeof(size_t)); i++) {
-        if (numOfValidArgs[i] == 0) {
-            LOG(ERROR) << "Failed reading parsed parameter " << (int)i << " ==> Abort";
-            return false;
-        }
-    }
-
-    //save average RSSI in watt
-    // ShortTermRSSIAverage values are space-separated:
-    for (const auto &rssi : beerocks::string_utils::str_split(ShortTermRSSIAverage, ' ')) {
-        float s_float = float(beerocks::string_utils::stoi(std::string(rssi)));
+    // Format "ShortTermRSSIAverage=%d %d %d %d"
+    auto samples = beerocks::string_utils::str_split(tmp_str, ' ');
+    for (const auto &s : samples) {
+        float s_float = float(beerocks::string_utils::stoi(s));
         if (s_float > beerocks::RSSI_MIN) {
             sta_stats.rx_rssi_watt += std::pow(10, s_float / float(10));
             sta_stats.rx_rssi_watt_samples_cnt++;
         }
     }
 
-    //save average SNR in watt
-    // SNR values are space-separated:
-    for (const auto &snr : beerocks::string_utils::str_split(SNR, ' ')) {
-        float s_float = float(beerocks::string_utils::stoi(std::string(snr)));
-        if (s_float > beerocks::SNR_MIN) {
+    // SNR
+    if (!read_param_str("SNR", reply, &tmp_str)) {
+        LOG(ERROR) << "Failed reading SNR parameter!";
+        return false;
+    }
+    // Format "SNR=%d %d %d %d"
+    samples = beerocks::string_utils::str_split(tmp_str, ' ');
+    for (const auto &s : samples) {
+        float s_float = float(beerocks::string_utils::stoi(s));
+        if (s_float >= beerocks::SNR_MIN) {
             sta_stats.rx_snr_watt += std::pow(10, s_float / float(10));
             sta_stats.rx_snr_watt_samples_cnt++;
         }
     }
 
-    // TODO: Update RSSI externally!
-    sta_stats.tx_phy_rate_100kb = (LastDataDownlinkRate / 100);
-    sta_stats.rx_phy_rate_100kb = (LastDataUplinkRate / 100);
-    calc_curr_traffic(BytesSent, sta_stats.tx_bytes_cnt, sta_stats.tx_bytes);
-    calc_curr_traffic(BytesReceived, sta_stats.rx_bytes_cnt, sta_stats.rx_bytes);
-    calc_curr_traffic(PacketsSent, sta_stats.tx_packets_cnt, sta_stats.tx_packets);
-    calc_curr_traffic(PacketsReceived, sta_stats.rx_packets_cnt, sta_stats.rx_packets);
+    // Last Downlink (TX) Rate
+    if (!read_param_int("LastDataDownlinkRate", reply, tmp_int)) {
```

Comment:
```suggestion
    int64_t LastDataDownlinkRate;
    if (!read_param_int("LastDataDownlinkRate", reply, LastDataDownlinkRate)) {
```

Reply:
Since it is duplicate/related to comment https://github.com/prplfoundation/prplMesh/pull/1491#discussion_r449626524.
Resolving.